zilliqa/src/state.rs

```diff
@@ -44,20 +46,25 @@ pub struct State {
     scilla_lib_dir: String,
     pub block_gas_limit: EvmGas,
     pub gas_price: u128,
+    pub zil_chain_id: u64,
```

Comment:
Why not store a `ChainId`? Then we should also remove the `chain_id` argument from some of the execution functions in `exec.rs` (e.g. `apply_transaction_evm`) and instead use `self.chain_id.eth`?